src/contracts/factory/FundFactory.ts

```diff
@@ -169,121 +150,208 @@ export class FundFactory extends Contract {
     ];
 
     const validate = async () => {
-      const hub = await this.getManagersToHubs(from);
-      if (!isZeroAddress(hub)) {
-        throw new FundSetupAlreadyStartedError();
+      this.validateHubNotSet(from);
+
+      const registry = new Registry(this.environment, await this.getRegistry());
+
+      // validation for beginSetup
+      const [isValidFundName, canUseFundName, denominationAssetRegistered] = await Promise.all([
+        registry.isValidFundName(settings.name),
+        registry.canUseFundName(from, settings.name),
+        registry.isAssetRegistered(settings.denominationAsset),
+      ]);
+      if (!isValidFundName) throw new InvalidFundNameError(settings.name);
```

Comment:
What did we say about single-line if() statements?

Reply:
Passive aggressive code reviews are the new chic.

Reply:
While I agree with you in general, it just looks nicer this way...